ark-client/src/lib.rs

```diff
@@ -289,6 +289,7 @@ where
         }
     }
 
+    // TODO: Add a timeout here. Maybe more generally.
```

Comment:
Do we want to tackle this now? Or maybe create a ticket for it and we do it later.

Reply:
I think it's probably better if we tackle this holistically. I'll open an issue.